truffle/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/InstrumentRegistrationProcessor.java

```diff
@@ -57,24 +57,34 @@ public SourceVersion getSupportedSourceVersion() {
         return SourceVersion.latest();
     }
 
+    private static int NumPropertiesPerEntry = 4;
+
     private void generateFile(List<TypeElement> instruments) {
         String filename = "META-INF/truffle/instrument";
         Properties p = new Properties();
+        int numInstruments = loadIfFileAlreadyExists(filename, p);
 
-        int cnt = 0;
         for (TypeElement l : instruments) {
             Registration annotation = l.getAnnotation(Registration.class);
             if (annotation == null) {
                 continue;
             }
-            String prefix = "instrument" + ++cnt + ".";
+
+            int instNum = findInstrument(annotation.id(), p);
+            if (instNum == 0) { // not found
+                numInstruments += 1;
+                instNum = numInstruments;
+            }
+
+            String prefix = "instrument" + instNum + ".";
             String className = processingEnv.getElementUtils().getBinaryName(l).toString();
+
             p.setProperty(prefix + "id", annotation.id());
             p.setProperty(prefix + "name", annotation.name());
             p.setProperty(prefix + "version", annotation.version());
             p.setProperty(prefix + "className", className);
         }
-        if (cnt > 0) {
+        if (numInstruments > 0) {
```

Comment:
Ehm, why?
My intent with this code was: 
 - search for a matching entry by `id`
 - if it is found, overwrite its current values

Why does that not work for changing a property for the annotation?